conflictlog/pool.go

```diff
@@ -218,12 +219,17 @@ func (pool *connPool) UpdateReapInterval(d time.Duration) {
 // Get returns an object from the pool. If there is none then it creates
 // one by calling newConnFn() and returns it. It is guaranteed that either
 // an error or a non-nil connection object will be returned
-func (pool *connPool) Get(bucketName string) (conn io.Closer, err error) {
+func (pool *connPool) Get(bucketName string, timeout time.Duration) (conn io.Closer, err error) {
 	if pool.closed {
 		err = ErrClosedConnPool
 		return
 	}
 
+	if !pool.limiter.AcquireWithTimeout(timeout) {
```

Comment:
Either have `AcquireWithTimeout` feed in a `finCh` to watch, or watch for `finCh` here to bail out early should `finCh` is closed prior to hitting timeout

And return correct error (i.e. `partStopped`) in that case